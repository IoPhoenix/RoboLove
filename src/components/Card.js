import React, {Component} from 'react';

class Card extends Component {

    changeCard = (e) => {
        console.log(this);
    }

    render() {
        return (
            <div onClick={this.changeCard} className='tc bg-light-green b--green dib br3 pa3 ma2 bw2 grow shadow-5 w-25'>
                <img src={`https://robohash.org/${this.props.id}?200x200`} width='200' height='200' alt={`robot named ${this.props.name}`} />
                <div>
                    <h2>{this.props.name}</h2>
                    <p>{this.props.email}</p>
                </div>
            </div>
        );
    }
}

export default Card;