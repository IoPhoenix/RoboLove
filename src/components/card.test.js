import { shallow, mount, render } from 'enzyme';
import React from 'react';
import Card from './Card';

// shallow: test only one component, no child components
// mount: test components that interact with dom, lifecycle methods, full rendering
// render: render React components in static html, uses cheerio, 
    // tests child components as well

it('expect to render Card component', () => {
    expect(shallow(<Card/>)).toMatchSnapshot();
});