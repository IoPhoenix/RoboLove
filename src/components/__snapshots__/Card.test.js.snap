// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="robot named undefined"
          height="200"
          src="https://robohash.org/undefined?200x200"
          width="200"
        />,
        <div
          className="robotInfo"
        >
          <h2
            className="mt3"
          />
          <p
            className="mt2"
          />
        </div>,
      ],
      "className": "flex flex-column items-center tc bg-light-green b--green dib br3 pa3 ma3 bw2 grow shadow-5 w-80 w-50-ns w-25-l",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "robot named undefined",
          "height": "200",
          "src": "https://robohash.org/undefined?200x200",
          "width": "200",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="mt3"
            />,
            <p
              className="mt2"
            />,
          ],
          "className": "robotInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "mt3",
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "mt2",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="robot named undefined"
            height="200"
            src="https://robohash.org/undefined?200x200"
            width="200"
          />,
          <div
            className="robotInfo"
          >
            <h2
              className="mt3"
            />
            <p
              className="mt2"
            />
          </div>,
        ],
        "className": "flex flex-column items-center tc bg-light-green b--green dib br3 pa3 ma3 bw2 grow shadow-5 w-80 w-50-ns w-25-l",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "robot named undefined",
            "height": "200",
            "src": "https://robohash.org/undefined?200x200",
            "width": "200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="mt3"
              />,
              <p
                className="mt2"
              />,
            ],
            "className": "robotInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "mt3",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "mt2",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
